turtlesim/tutorials/teleop_turtle_key.cpp

```diff
@@ -81,55 +80,58 @@ void TeleopTurtle::keyLoop()
   puts("---------------------------");
   puts("Use arrow keys to move the turtle.");
 
-
   for(;;)
   {
     // get the next event from the keyboard  
-    if(read(kfd, &c, 1) < 0)
+    if(read(kfd, &c, sizeof(c)) < 0)
     {
       perror("read():");
       exit(-1);
     }
 
     linear_=angular_=0;
-    ROS_DEBUG("value: 0x%02X\n", c);
+    RCLCPP_DEBUG(this->get_logger(), "value: 0x%02X\n", c);
   
     switch(c)
     {
       case KEYCODE_L:
-        ROS_DEBUG("LEFT");
+        RCLCPP_DEBUG(this->get_logger(), "LEFT");
         angular_ = 1.0;
         dirty = true;
         break;
       case KEYCODE_R:
-        ROS_DEBUG("RIGHT");
+        RCLCPP_DEBUG(this->get_logger(), "RIGHT");
         angular_ = -1.0;
         dirty = true;
         break;
       case KEYCODE_U:
-        ROS_DEBUG("UP");
+        RCLCPP_DEBUG(this->get_logger(), "UP");
         linear_ = 1.0;
         dirty = true;
         break;
       case KEYCODE_D:
-        ROS_DEBUG("DOWN");
+        RCLCPP_DEBUG(this->get_logger(), "DOWN");
         linear_ = -1.0;
         dirty = true;
         break;
+      case KEYCODE_S:
+        RCLCPP_DEBUG(this->get_logger(), "STOP");
+        linear_ = 0.0;
+        angular_ = 0.0;
+        dirty = true;
+        break;
```

Comment:
Please do not introduce new features in the same PR which performs the porting. Please contribute that separate to the default branch.

Reply:
Accepted